File: assignment_2/ADTs/MyStack.java
package ADTs;

import Exceptions.EmptyContainerException;

import java.util.Stack;

public class MyStack<T> implements IStack<T> {
    private Stack<T> stack;

    public MyStack(){
        this.stack = new Stack<>();
    }

    public void push(T elem) {
        stack.push(elem);
    }

    public T pop() throws EmptyContainerException {
        if(this.stack.isEmpty())
            throw new EmptyContainerException("THE STACK IS EMPTY!");
        return stack.pop();
    }

    public boolean isEmpty() {
        return stack.empty();
    }

    public String toString() {
        StringBuilder result = new StringBuilder();
        int size = 100;
        result.setLength(size);
        if(this.stack.isEmpty()){
            result.append("[ ]");
            return result.toString();
        }
        int i = this.stack.size() - 1;
        result.append("[");
        result.append(this.stack.get(i).toString());
        i--;
        while(i >= 0) {
            result.append(" | ");
            if(result.length() > size) {
                size = 2 * size;
                result.setLength(size);
            }
            result.append(this.stack.get(i).toString());
            i--;
        }
        result.append("]");
        return result.toString();

    }
}

File: Toy_Language_Interpreter_C/ADTs/MyStack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace a7
{
    class MyStack<T> : IStack<T>
    {
        private Stack<T> stack;
        public Stack<T> Stack
        {
            get
            {
                return this.stack;
            }
        }

        public MyStack()
        {
            this.stack = new Stack<T>();
        }

        public void Push(T elem)
        {
            this.stack.Push(elem);
        }

        public T Pop()
        {
            if (this.IsEmpty())
                throw new EmptyStackException();
            return this.stack.Pop(); 
        }

        public Boolean IsEmpty()
        {
            if (this.stack.ToArray().Length == 0)
                return true;
            return false;
        }

        public override string ToString()
        { 
            String result = "";
            T[] temp = this.stack.ToArray();
            foreach(T elem in temp)
            {
                result = result + elem.ToString() + "\n";
            }
            return result;
        }


    }
}
